src/include/sof/audio/data_blob.h

```diff
@@ -66,9 +74,35 @@ int comp_data_blob_get_cmd(struct comp_data_blob_handler *blob_handler,
 /**
  * Returns new data blob handler.
  *
+ * Data blob handler has two modes of operation, single_blob == false
+ * stands for double blob mode that allow seamless configuration
+ * updates on the fly. In single_blob == true mode there is at most one
+ * blob allocated at one time and configuration update is not allowed
+ * if the component is active.
+ *
+ * @param dev Component device
+ * @param single_blob Set true for single configuration blob operation
+ * @param alloc Optional blob memory allocator function pointer
+ * @param free Optional blob memory free function pointer
+ */
+struct comp_data_blob_handler *
+comp_data_blob_handler_new_full(struct comp_dev *dev, bool single_blob,
+				void *(*alloc)(size_t size),
```

Comment:
Rename to comp_data_blob_handler_new_ext, to point it as an extension to original handler.